fs-base/src/main/java/com/palantir/giraffe/file/MoreFiles.java

```diff
@@ -569,6 +570,55 @@ public static boolean isEmpty(Path path) throws IOException {
     }
 
     /**
+     * A {@code FileVisitor} that builds a list of {@code Path} objects it visits.
+     *
+     * @author jyu
+     */
+    private static final class ListVisitor extends SimpleFileVisitor<Path> {
+        private final ImmutableList.Builder<Path> paths;
+
+        public ListVisitor() {
+            paths = new ImmutableList.Builder<>();
+        }
+
+        /**
+         * Adds the visited {@code Path} to the list of visited {@code Path} objects.
+         */
+        @Override
+        public FileVisitResult visitFile(Path file, BasicFileAttributes attrs)
+                throws IOException {
+            paths.add(file);
+            return FileVisitResult.CONTINUE;
+        }
+
+        /**
+         * Returns the current list of visited {@code Path} objects.
+         *
+         * The returned path list can change between calls if this visitor visits any paths.
+         */
+        public List<Path> getPaths() {
+            return paths.build();
+        }
+    }
+
+    /**
+     * Gets a list of the files in the given directory and all sub-directories.
+     * The returned paths are relative to the given directory.
```

Comment:
Oops, I totally lied: I was returning absolute paths.

`[/Library/Java/JavaVirtualMachines/jdk1.7.0_71.jdk/Contents/Home/jre/bin/java, /Library/Java/JavaVirtualMachines/jdk1.7.0_71.jdk/Contents/Home/jre/bin/keytool, ...`

I'll patch this real quick.